docs/integration_tests.md

```diff
@@ -0,0 +1,76 @@
+# Integration tests
+
+Integration tests are useful to ensure that the code works well for any use cases.
+
+They can also be used as sample on how to use the Matrix SDK.
+
+In a ideal world, every API of the SDK should be covered by integration tests. For the moment, we have test mainly for the Crypto part, which is the tricky part. But it covers quite a lot of features: accounts creation, login to existing account, send encrypted messages, keys backup, verification, etc.
+
+The Matrix SDK is able to open multiple sessions, for the same user, of for different users. This way we can test communication between several sessions on a single device.
+
+## Pre requirements
+
+Integration tests need a homeserver running on localhost.
+
+The documentation describe what we do to have one, using [Synapse](https://github.com/matrix-org/synapse/), which is the Matrix reference homeserver.
+
+## Install and run Synapse
+
+Steps:
+
+- Install virtual env
+
+```bash
+python3 -m pip install virtualenv
+```
+
+- clone Synapse repository
+
+```bash
+git clone git@github.com:matrix-org/synapse.git
```

Comment:
It will give the error:
git@github.com: Permission denied (publickey).

Should we use git clone https://github.com/matrix-org/synapse.git instead?